torchtext/experimental/datasets/raw/common.py

```diff
@@ -25,6 +27,87 @@ def wrap_datasets(datasets, split):
     return datasets
 
 
+def dataset_docstring_header(fn):
+    argspec = inspect.getfullargspec(fn)
+    if not (argspec.args[0] == "root" and
+            argspec.args[1] == "split" and
+            argspec.args[2] == "offset"):
+        raise ValueError("Internal Error: Given function {} did not adhere to standard signature.".format(fn))
+    default_split = argspec.defaults[1]
+
+    if isinstance(default_split, tuple):
+        example_subset = default_split[:2]
+        if len(default_split) < 3:
+            example_subset = (default_split[1],)
+        return """{} dataset
+
+        Separately returns the {} split
+
+        Args:
+            root: Directory where the datasets are saved.
+                Default: ".data"
+            split: split or splits to be returned. Can be a string or tuple of strings.
+                By default, all three datasets are generated. Users
+                could also choose any subset of them, for example {} or just 'train'.
+                Default: {}
+            offset: the number of the starting line.
+                Default: 0
+        """.format(fn.__name__, "/".join(default_split), str(example_subset), str(default_split)) + fn.__doc__
+
+    if isinstance(default_split, str):
+        return """{} dataset
+
+        Only returns the {default_split} split
+
+        Args:
+            root: Directory where the datasets are saved.
+                Default: ".data"
+            split: Only {default_split} is available.
+                Default: {default_split}
+            offset: the number of the starting line.
+                Default: 0
+        """.format(fn.__name__, default_split=default_split) + fn.__doc__
+
+    raise ValueError("default_split type expected to be of string or tuple but got {}".format(type(default_split)))
+
+
+def dataset_docstring_header_decorator(fn):
+    fn.__doc__ = dataset_docstring_header(fn)
+    return fn
+
+
+def input_sanitization_decorator(fn):
+    argspec = inspect.getfullargspec(fn)
+    if not (argspec.args[0] == "root" and
+            argspec.args[1] == "split" and
+            argspec.args[2] == "offset" and
+            argspec.defaults[0] == ".data" and
+            argspec.defaults[2] == 0 and
+            argspec.varargs is None and
+            argspec.varkw is None and
+            len(argspec.kwonlyargs) == 0 and
+            argspec.kwonlydefaults is None and
+            len(argspec.annotations) == 0
+            ):
+        raise ValueError("Internal Error: Given function {} did not adhere to standard signature.".format(fn))
+
```

Comment:
The following part is not intuitively clear what it intends to do, so adding comments will be helpful.